### PR TITLE
LC-405: lucille opensearch ingest example

### PR DESCRIPTION
**Pull request by Jozurf**
Notes:
- sendEnabled set to false for further debugging
- no test created for example
- will need to change the enron folder path in the opensearch.conf in the conf folder
- if need to test on Opensearch server, change the url in opensearch config to appropriate username and password

### CODE REVIEW

File: lucille-examples/lucille-opensearch-ingest-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    <dependency>
+      <groupId>com.kmwllc</groupId>
+      <artifactId>lucille-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.kmwllc</groupId>
+      <artifactId>lucille-tika</artifactId>
+    </dependency>
+    <!-- https://mvnrepository.com/artifact/mysql/mysql-connector-java -->
+<dependency>
```
Comment (rseitz) [inline lucille-examples/lucille-opensearch-ingest-example/pom.xml:41]: minor: pls. fix indentation here

File: lucille-examples/lucille-opensearch-ingest-example/conf/opensearch.conf [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+       class : "com.kmwllc.lucille.stage.DeleteFields"
+       fields : ["file_modification_date", "file_creation_date", "file_size_bytes", "file_content", "_message_from_name"]
+     }
+    ]
+  }
+]
+
+indexer {
+  type: "OpenSearch"
+  class: "com.kmwllc.lucille.indexer.OpenSearchIndexer"
```
Comment (rseitz) [inline lucille-examples/lucille-opensearch-ingest-example/conf/opensearch.conf:54]: for now, the class is not needed when providing a type

File: lucille-examples/lucille-opensearch-ingest-example/conf/opensearch.conf [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  batchTimeout: 1000
+  batchSize: 2000
+  # idOverrideField: "message_id"
```
Comment (rseitz) [inline lucille-examples/lucille-opensearch-ingest-example/conf/opensearch.conf:57]: add comment to explain why someone would want to use this

**Comment by rseitz** [comment]
Add readme explaining 1) what this example does, 2) where to get the enron dataset, 2) how to set up environment variable or edit lucille config, 3) how to set up opensearch (including any required mappings), 4) how to run lucille.

File: lucille-examples/lucille-opensearch-ingest-example/pom.xml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    <dependency>
+      <groupId>com.kmwllc</groupId>
+      <artifactId>lucille-core</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.kmwllc</groupId>
+      <artifactId>lucille-tika</artifactId>
+    </dependency>
+    <!-- https://mvnrepository.com/artifact/mysql/mysql-connector-java -->
+    <dependency>
```
Comment (rseitz) [inline lucille-examples/lucille-opensearch-ingest-example/pom.xml:41]: Is mysql needed for this example?

File: lucille-examples/lucille-opensearch-ingest-example/conf/tika-config.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<properties>
+    <parsers>
+      <parser class="org.apache.tika.parser.DefaultParser"/>
```
Comment (rseitz) [inline lucille-examples/lucille-opensearch-ingest-example/conf/tika-config.xml:4]: Would we get any benefit from not including DefaultParser here? Or from including it but putting the RFC parser first?

Comment (Jozurf) [reply lucille-examples/lucille-opensearch-ingest-example/conf/tika-config.xml:4]: DefaultParser is needed for parsing the body, RFC parser is need for the rest of the metadata

**Review by rseitz — APPROVED** [review]

**Pull request merged**